modules/token/keeper/evm_hook.go

```diff
@@ -0,0 +1,106 @@
+package keeper
+
+import (
+	"math/big"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ethereum/go-ethereum/core"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/irisnet/irismod/contracts"
+	"github.com/irisnet/irismod/modules/token/types"
+)
+
+type erc20Hook struct {
+	k Keeper
+}
+
+// PostTxProcessing processes the transaction receipt for ERC20 token swap to native.
+//
+// Parameters:
+//
+//	ctx: the context in which the function is executed.
+//	msg: the core message associated with the transaction.
+//	receipt: the Ethereum receipt containing the transaction logs.
+//
+// Return type: error
+func (hook erc20Hook) PostTxProcessing(ctx sdk.Context, msg core.Message, receipt *ethtypes.Receipt) error {
+	erc20 := contracts.ERC20TokenContract.ABI
+	for _, log := range receipt.Logs {
+		// Note: the `SwapToNative` event contains 1 topics
+		//  SwapToNative(address from, string to, uint256 amount)
+		if len(log.Topics) != 1 {
+			continue
+		}
+
+		// Check if event is included in ERC20
+		eventID := log.Topics[0]
+		event, err := erc20.EventByID(eventID)
+		if err != nil {
+			continue
```

Comment:
What are the err scenarios here? Do we need to print err logs and roll back transactions?

Reply:
We only care about the events defined in our erc20 contract. We don’t have to worry about the user’s contract events, so even if there is an error in parsing, we don’t need to care.